File: exercise/python/day23/app/main.py
from core.control import ControlSystem, ReindeersNeedRestException, SleighNotStartedException


def main():
    control_system = ControlSystem()
    control_system.start_system()

    keep_running = True

    while keep_running:
        command = input("Enter a command (ascend (a), descend (d), park (p), or quit (q)): ")

        if command in ("ascend", "a"):
            try:
                control_system.ascend()
            except (ReindeersNeedRestException, SleighNotStartedException) as e:
                print(e)
        elif command in ("descend", "d"):
            try:
                control_system.descend()
            except SleighNotStartedException as e:
                print(e)
        elif command in ("park", "p"):
            try:
                control_system.park()
            except SleighNotStartedException as e:
                print(e)
        elif command in ("quit", "q"):
            keep_running = False
        else:
            print("Invalid command. Please try again.")

    control_system.stop_system()


if __name__ == "__main__":
    main()

File: exercise/ts/day23/src/app/main.ts
import {ControlSystem} from "../control.core/controlSystem";
import {ReindeersNeedRestException, SleighNotStartedException} from "../control.core/exceptions";

const controlSystem = new ControlSystem();
controlSystem.startSystem();

const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout
});

function askQuestion() {
    readline.question("Enter a command (ascend (a), descend (d), park (p), or quit (q)): ", (command: string) => {
        switch (command) {
            case "ascend":
            case "a":
                try {
                    controlSystem.ascend();
                } catch (e) {
                    if (e instanceof ReindeersNeedRestException || e instanceof SleighNotStartedException) {
                        console.log(e.message);
                    }
                }
                break;
            case "descend":
            case "d":
                try {
                    controlSystem.descend();
                } catch (e) {
                    if (e instanceof SleighNotStartedException) {
                        console.log(e.message);
                    }
                }
                break;
            case "park":
            case "p":
                try {
                    controlSystem.park();
                } catch (e) {
                    if (e instanceof SleighNotStartedException) {
                        console.log(e.message);
                    }
                }
                break;
            case "quit":
            case "q":
                readline.close();
                controlSystem.stopSystem();
                return;
            default:
                console.log("Invalid command. Please try again.");
        }
        askQuestion();
    });
}

askQuestion();
